lib/kafka/monitor.rb

```diff
@@ -76,5 +78,27 @@ def track_submission_exhaustion(msg, topic, payload)
       track_request('error', "Kafka::EventBusSubmissionJob for #{topic} exhausted!",
                     "#{STATSD_KEY_PREFIX}.exhausted", call_location:, **additional_context)
     end
+
+    private
+
+    def redact_icn(hash)
+      return hash unless hash.is_a?(Hash)
+
+      stack = [hash]
+
+      while (current = stack.pop)
+        current.each do |key, value|
+          if key.upcase == 'ICN'
+            current[key] = '[REDACTED]'
+          elsif value.is_a?(Hash)
+            stack.push(value)
+          elsif value.is_a?(Array)
+            value.each { |item| stack.push(item) if item.is_a?(Hash) }
+          end
+        end
+      end
+
+      hash
+    end
```

Comment:
I checked with DSVA support and they do filter icn and PII but only on incoming params, not payloads being loagged .This might be overkill logic but I was having trouble finding any built in methods like `deep_transform_values` that could work. We could always just check the first two layers of of hashes for an ICN field.

Reply:
While this is an algorithmically correct way to check for a key in a potentially deeply nested hash, I think for our case we could keep it simple and just check which topic we're dealing with. I also thought this `#redact_icn` method should like in the `Kafka::Topic` concern instead of here so we can use it more universally. The `Kafka::Monitor` method should get the redacted payload, similarly how they are getting the final topic name. Does that make sense?

Reply:
I moved the redact_icn to the concern but i figured there may be a edge case where if they choose the primary topic, if they malformed it, the payload in the error log would still need to be redacted regardless of the topic choose. If we want to assume they'll get the schema correct though (especially if they use the `submit_event` helper method), I'm good with just changing it to look for icn in the first layer of the hash, is that what you were thinking?

Reply:
I'm okay with leaving the `#redact_icn` method  to be more generic. It's true that this version would be able to handle malformed payloads in addition to correct ones. Let's leave it as you have it, even thought the risk of malformed payloads ending up in logs is currently small but that can change over time. The `Kafka.submit_event` and the `EventBusSubmissionJob` can both be trusted to pick the right topic and redact the payload in the error logs. If anyone uses the `Kafka::AvroProducer` directly, that method would not log payloads at all.